torch/_subclasses/meta_utils.py

```diff
@@ -433,15 +436,49 @@ def empty_create(inner_t, inner_src):
                         # their sizes will be used to construct the (symbolic) sizes of the wrapper tensor.
                         from torch._dynamo.source import AttrSource
 
-                        r = transform_subclass(
-                            t,
-                            lambda attr, inner_t: callback(
-                                lambda: empty_create(
-                                    inner_t,
-                                    AttrSource(source, attr),
+                        if t.is_nested:
+                            # Avoid circular import
+                            from torch._dynamo.source import (
+                                TensorProperty,
+                                TensorPropertySource,
+                            )
+
+                            # For nested tensors, manually do transform_subclass
+                            # so we can insert some special processing on ctx
+                            attrs, ctx = t.__tensor_flatten__()
+                            transformed_tensors_dict = {}
+                            for attr in attrs:
+                                inner_t = getattr(t, attr)
+                                transformed_tensors_dict[attr] = callback(
+                                    lambda: empty_create(
+                                        inner_t, AttrSource(source, attr)
+                                    )
                                 )
-                            ),
-                        )
+                            # We expect JaggedTensor to have a 'ragged_size' in
+                            # its context
+                            assert isinstance(ctx, dict) and "ragged_size" in ctx
+                            # Replace the eager ragged size with our freshly
+                            # allocated jagged size that has a source
+                            ctx["ragged_size"] = shape_env.create_symintnode(
+                                shape_env.create_symbol(
+                                    t._size[1],
+                                    TensorPropertySource(
+                                        source, TensorProperty.SIZE, 1
+                                    ),
+                                ),
+                                hint=t._size[1],
```

Comment:
`t._size[1]` is the eager mode ragged size here (holding a SymNode)? Maybe we should assert that?